src/actions/reportsActions.js

```diff
@@ -139,6 +140,23 @@ export const unfavoriteReportACreator = reportId => {
   }
 }
 
+export const downloadReportZipFile = (downloadReportZipFileStatus, downloadReportZipFileErrors) => {
+  return { type: DOWNLOAD_REPORT_ZIP_FILE, downloadReportZipFileStatus, downloadReportZipFileErrors}
+};
+
+export const downloadReportZipFileACreator = (fileName) => dispatch => {
+  return new Promise((resolve, reject) => {
+    WebApi.reports.get.reportZip(fileName).then(response => {
+      dispatch(downloadReportZipFile(true, []));
+      window.open(response.replyBlock.request.responseURL);
+      resolve();
```

Comment:
What is a reason of that resolve? We already have got returned data from api by promise.

Reply:
resolve method give signal of successfull end of promise. It doesn't necessarily have to return something